File: hello-dyn-java/BasicCalculator.java
// 基础计算器实现
public class BasicCalculator implements Calculator {
    @Override
    public double addFn(double a, double b) {
        return a + b;
    }
    
    @Override
    public double subFn(double a, double b) {
        return a - b;
    }
    
    @Override
    public double mulFn(double a, double b) {
        return a * b;
    }
    
    @Override
    public double divFn(double a, double b) {
        if (b == 0) {
            throw new ArithmeticException("除数不能为零");
        }
        return a / b;
    }
}

File: hello-dyn-python/basic_calculator.py
from calculator_base import Calculator


class BasicCalculator(Calculator):
    """基础计算器实现"""

    def addFn(self, a, b):
        return a + b

    def subFn(self, a, b):
        return a - b

    def mulFn(self, a, b):
        return a * b

    def divFn(self, a, b):
        if b == 0:
            raise ValueError("除数不能为零")
        return a / b
